sample-authn/src/main/java/net/trajano/ms/example/authn/AuthnResource.java

```diff
@@ -1,87 +1,71 @@
-import net.trajano.ms.common.oauth.AllowAnyClientValidator;
-import net.trajano.ms.common.oauth.BaseTokenResource;
-import net.trajano.ms.common.oauth.GrantHandler;
-import net.trajano.ms.common.oauth.GrantTypes;
+import net.trajano.ms.auth.token.GrantTypes;
 
+/**
+ * This works like the FORM based login of Java EE. It allows any user name as
+ * long as the password is "password"
+ */
 @SwaggerDefinition(
-    securityDefinition = @SecurityDefinition(basicAuthDefinitions = @BasicAuthDefinition(key = "client",
-        description = "Client ID/Secret")),
     info = @Info(
         title = "Sample Authn Microservice",
         version = "1.0"))
-@Api(tags = "unprotected")
 @Component
 @Path("/authn")
 @PermitAll
-public class AuthnResource extends BaseTokenResource {
+public class AuthnResource {
 
-    @Autowired
-    public AuthnResource(
-        final List<GrantHandler> grantHandlers) {
+    @Context
+    private Client client;
 
-        super(new AllowAnyClientValidator(), grantHandlers);
-    }
+    @Autowired
+    private CryptoOps cryptoOps;
 
     @POST
-    @Consumes(MediaType.APPLICATION_JSON)
-    public Response json(@Context final HttpHeaders httpHeaders,
-        final AuthnRequest req) {
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    @Produces(MediaType.APPLICATION_JSON)
+    public OAuthTokenResponse json(@FormParam("j_username") String username,
+        @FormParam("j_password") String password,
+        @HeaderParam(HttpHeaders.AUTHORIZATION) String authorization) {
 
-        final MultivaluedMap<String, String> form = new MultivaluedHashMap<>();
-        form.putSingle("grant_type", req.getGrantType());
-        form.putSingle("username", req.getUsername());
-        form.putSingle("password", req.getPassword());
-        return super.token(httpHeaders, form);
-    }
+        if (!"password".equals(password)) {
+            throw OAuthTokenResponse.unauthorized(ErrorCodes.UNAUTHORIZED_CLIENT, "invalid username/password combination", "FORM");
+        }
+        final JwtClaims claims = new JwtClaims();
+        claims.setSubject(username);
 
-    @ApiImplicitParams({
-        @ApiImplicitParam(name = "grant_type",
-            value = "Grant type",
-            required = true,
-            dataType = "java.lang.String",
-            example = GrantTypes.PASSWORD,
-            paramType = "form"),
-        @ApiImplicitParam(name = "username",
-            value = "Username",
-            dataType = "java.lang.String",
-            required = true,
-            paramType = "form"),
-        @ApiImplicitParam(name = "password",
-            value = "Password",
-            dataType = "java.lang.String",
-            required = true,
-            paramType = "form")
-    })
-    @Override
-    public Response token(final HttpHeaders httpHeaders,
-        final MultivaluedMap<String, String> form) {
+        String[] auth = HttpAuthorizationHeaders.parseBasicAuthorization(authorization);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "auth". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "auth" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1481)